gems/core/tasks.py

```diff
@@ -235,3 +237,65 @@ def sync_group_members(self):
     api = ContactsApiClient(settings.VUMI_TOKEN)
     for member in unsynced_members:
         process_group_member(api, member.contact, member.group)
+
+
+def date_construct_helper(today, delta):
+    t = today
+
+    if delta is not None:
+        t = today - delta
+
+    t = t.replace(hour=0, minute=0, second=0, microsecond=0)
+    return t
+
+
+@task
+def fetch_total_sent_smses():
+    logger.info('Fetching TOTAL SENT SMSES::STARTED')
+    api = MetricsApiClient(settings.VUMI_TOKEN)
+    surveys = Survey.objects.values('survey_id').distinct()
+    totals = [0, 0, 0, 0, 0, 0, 0]
+    today = datetime.now()
+    total_dates = [
+        date_construct_helper(today, timedelta(days=6)),
+        date_construct_helper(today, timedelta(days=5)),
+        date_construct_helper(today, timedelta(days=4)),
+        date_construct_helper(today, timedelta(days=3)),
+        date_construct_helper(today, timedelta(days=2)),
+        date_construct_helper(today, timedelta(days=1)),
+        date_construct_helper(today, None)
+    ]
+
+    for survey in surveys:
+        metric = 'conversations.' + survey[str('survey_id')] + '.outbound_unique_addresses.avg'
```

Comment:
@hodgestar: survey['survey_id']: dictionary doesn't contain a key called 'survey_id', hence the reason for the str() cast.